src/user.rs

```diff
@@ -50,73 +52,66 @@ pub(crate) async fn user_main(
 
     // If normal tasks are defined, loop launching tasks until parent tells us to stop.
     if !thread_task_set.weighted_tasks.is_empty() {
-        'launch_tasks: loop {
-            // Tracks the time it takes to loop through all GooseTasks when Coordinated Omission
-            // Mitigation is enabled.
-            thread_user.update_request_cadence(thread_number).await;
-
-            for (thread_task_index, thread_task_name) in &thread_task_set.weighted_tasks {
-                // Determine which task we're going to run next.
-                let function = &thread_task_set.tasks[*thread_task_index].function;
-                debug!(
-                    "launching on_start {} task from {}",
-                    thread_task_name, thread_task_set.name
-                );
-                // Invoke the task function.
-                let _todo = invoke_task_function(
-                    function,
-                    &mut thread_user,
-                    *thread_task_index,
-                    thread_task_name,
-                )
-                .await;
-
-                // Prepare to sleep for a random value from min_wait to max_wait.
-                let wait_time = if thread_user.max_wait > 0 {
-                    rand::thread_rng().gen_range(thread_user.min_wait..thread_user.max_wait)
-                } else {
-                    0
-                };
-
-                // Counter to track how long we've slept, waking regularly to check for messages.
-                let mut slept: usize = 0;
-
-                // Wake every second to check if the parent thread has told us to exit.
-                let mut in_sleep_loop = true;
-                // Track the time slept for Coordinated Omission Mitigation.
-                let sleep_timer = time::Instant::now();
-                while in_sleep_loop {
-                    let mut message = thread_receiver.try_recv();
-                    while message.is_ok() {
-                        match message.unwrap() {
-                            // Time to exit, break out of launch_tasks loop.
-                            GooseUserCommand::Exit => {
-                                break 'launch_tasks;
-                            }
-                            command => {
-                                debug!("ignoring unexpected GooseUserCommand: {:?}", command);
-                            }
-                        }
-                        message = thread_receiver.try_recv();
+        let mut task_iter = thread_task_set.weighted_tasks.iter().cycle();
+        let next_task = Fuse::terminated();
```

Comment:
Please name this something that better documents what this variable is for, such as `time_to_next_task`. (I initially assumed it held a GooseTask from the name.)